<a href="https://colab.research.google.com/github/itsGab/previsao_uti_em_covid/blob/main/notebook/pred_covid_uti_0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Importando os dados de um [repositório](https://github.com/itsGab/previsao_uti_em_covid/) no Github 

In [1]:
data = "https://github.com/itsGab/previsao_uti_em_covid/blob/main/dados/Kaggle_Sirio_Libanes_ICU_Prediction.xlsx?raw=true"

Importando [pandas](https://pandas.pydata.org/docs/index.html), a biblioteca principal para tratamento e visualização dos dados

In [2]:
import pandas as pd

Lendas os dados a primeira vez e imprimindo as 5 primeiras linhas

In [3]:
df_inicial = pd.read_excel(data)
df_inicial.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65  ...    WINDOW  ICU
0                         0            1  ...       0-2    0
1                         0            1  ...       2-4    0
2                         0            1  ...       4-6    0
3                         0            1  ...      6-12    0
4                         0            1  ...  ABOVE_12    1

[5 rows x 231 columns]

Mostrando informações do DataFrame

In [4]:
df_inicial.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1925 entries, 0 to 1924
Columns: 231 entries, PATIENT_VISIT_IDENTIFIER to ICU
dtypes: float64(225), int64(4), object(2)
memory usage: 3.4+ MB


Podemos perceber que o DataFrame tem 1925 linhas e 231 colunas

Começando na colunas "PATIENT_VISIT_IDENTIFIER" e indo até "ICU"

A maior parte dos dados são no formato float (número decimal), possuindo alguns int (número inteiro) e apenas 2 object (que provavelmente são texto).

Chamando a função describe para termos uma idéia de como estão os nossos dados númericos (float ou int)

In [13]:
pd.set_option('display.max_columns', None)
df_inicial.describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65       GENDER  DISEASE GROUPING 1  \
count               1925.000000  1925.000000  1925.000000         1920.000000   
mean                 192.000000     0.467532     0.368831            0.108333   
std                  111.168431     0.499074     0.482613            0.310882   
min                    0.000000     0.000000     0.000000            0.000000   
25%                   96.000000     0.000000     0.000000            0.000000   
50%                  192.000000     0.000000     0.000000            0.000000   
75%                  288.000000     1.000000     1.000000            0.000000   
max                  384.000000     1.000000     1.000000            1.000000   

       DISEASE GROUPING 2  DISEASE GROUPING 3  DISEASE GROUPING 4  \
count         1920.000000         1920.000000         1920.000000   
mean             0.028125            0.097917            0.019792   
std              0.165373            0.297279            0.139320   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE GROUPING 5  DISEASE GROUPING 6          HTN  IMMUNOCOMPROMISED  \
count         1920.000000         1920.000000  1920.000000        1920.000000   
mean             0.128125            0.046875     0.213021           0.158333   
std              0.334316            0.211426     0.409549           0.365148   
min              0.000000            0.000000     0.000000           0.000000   
25%              0.000000            0.000000     0.000000           0.000000   
50%              0.000000            0.000000     0.000000           0.000000   
75%              0.000000            0.000000     0.000000           0.000000   
max              1.000000            1.000000     1.000000           1.000000   

             OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  ALBUMIN_MAX  \
count  1920.000000      821.000000    821.000000   821.000000   821.000000   
mean      0.809896        0.528527      0.528527     0.528527     0.528527   
std       0.392485        0.224100      0.224100     0.224100     0.224100   
min       0.000000       -1.000000     -1.000000    -1.000000    -1.000000   
25%       1.000000        0.605263      0.605263     0.605263     0.605263   
50%       1.000000        0.605263      0.605263     0.605263     0.605263   
75%       1.000000        0.605263      0.605263     0.605263     0.605263   
max       1.000000        1.000000      1.000000     1.000000     1.000000   

       ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  BE_ARTERIAL_MIN  \
count         821.0          821.000000        821.000000       821.000000   
mean           -1.0           -0.963433         -0.963433        -0.963433   
std             0.0            0.160870          0.160870         0.160870   
min            -1.0           -1.000000         -1.000000        -1.000000   
25%            -1.0           -1.000000         -1.000000        -1.000000   
50%            -1.0           -1.000000         -1.000000        -1.000000   
75%            -1.0           -1.000000         -1.000000        -1.000000   
max            -1.0            1.000000          1.000000         1.000000   

       BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  BE_VENOUS_MEAN  \
count       821.000000             821.0        821.000000      821.000000   
mean         -0.963433              -1.0         -0.931121       -0.931121   
std           0.160870               0.0          0.169509        0.169509   
min          -1.000000              -1.0         -1.000000       -1.000000   
25%          -1.000000              -1.0         -1.000000       -1.000000   
50%          -1.000000              -1.0         -1

Com base nas colunas e seus dados, foi percebido que alguns dados são categoricos de sim ou não, se a pessoa possui um caracteristica ou não, e assim foi separado essas colunas, representadas pela variável `colunas_categoricas`.

In [6]:
colunas_categoricas = ['PATIENT_VISIT_IDENTIFIER', 'AGE_ABOVE65', 
                       'AGE_PERCENTIL', 'GENDER', 'DISEASE GROUPING 1', 
                       'DISEASE GROUPING 2', 'DISEASE GROUPING 3', 
                       'DISEASE GROUPING 4', 'DISEASE GROUPING 5', 
                       'DISEASE GROUPING 6', 'HTN', 'IMMUNOCOMPROMISED', 
                       'OTHER', 'WINDOW', 'ICU']

Removendo as colunas categoricas de todas as colunas, foi possível separar as colunas com dados continuas, representada pela variável `colunas_continuas`

In [7]:
colunas_continuas = [coluna for coluna in df_inicial.columns if coluna not in colunas_categoricas]

## *** Hipótese 1: 
As colunas com dados continuos estão dividas em, 
* 'MEDIAN', 
* 'MEAN', 
* 'MIN', 
* 'MAX', 
* 'DIFF', 
* 'DIFF_REL'

Acredito não haver grande variação entre elas que façam dessas colunas relevante, então irei deixar apenas a 
* 'MEAN'

### testar a hipóteses!!!!

Com uma análise superficial do data frame, foi notado que não a uma variação muito exprecisava entre as varias medidas da mesmo "exame" (a max, min, median, mean, diff), principalmente levando em conta que estaremos usando para o modelo de previsão somente a primeira janela (window)

In [8]:
palavra = 'MEAN'
colunas_continuas_mean = [col for col in colunas_continuas if palavra in col]

aqui juntei as colunas categoricas a continuas com apenas o "mean" e representei a variavel com `colunas_restante`



In [9]:
colunas_restante = colunas_categoricas + colunas_continuas_mean

defini o DataFrame atual de trabalho como `df_atual`

In [10]:
df_atual = df_inicial[colunas_restante]

In [14]:
df_atual.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
0                         0            1          60th       0   
1                         0            1          60th       0   
2                         0            1          60th       0   
3                         0            1          60th       0   
4                         0            1          60th       0   

   DISEASE GROUPING 1  DISEASE GROUPING 2  DISEASE GROUPING 3  \
0                 0.0                 0.0                 0.0   
1                 0.0                 0.0                 0.0   
2                 0.0                 0.0                 0.0   
3                 0.0                 0.0                 0.0   
4                 0.0                 0.0                 0.0   

   DISEASE GROUPING 4  DISEASE GROUPING 5  DISEASE GROUPING 6  HTN  \
0                 0.0                 1.0                 1.0  0.0   
1                 0.0                 1.0                 1.0  0.0   
2                 0.0                 1.0                 1.0  0.0   
3                 0.0                 1.0                 1.0  0.0   
4                 0.0                 1.0                 1.0  0.0   

   IMMUNOCOMPROMISED  OTHER    WINDOW  ICU  ALBUMIN_MEAN  BE_ARTERIAL_MEAN  \
0                0.0    1.0       0-2    0           NaN               NaN   
1                0.0    1.0       2-4    0           NaN               NaN   
2                0.0    1.0       4-6    0      0.605263         -1.000000   
3                0.0    1.0      6-12    0           NaN               NaN   
4                0.0    1.0  ABOVE_12    1      0.000000         -0.871658   

   BE_VENOUS_MEAN  BIC_ARTERIAL_MEAN  BIC_VENOUS_MEAN  BILLIRUBIN_MEAN  \
0             NaN                NaN              NaN              NaN   
1             NaN                NaN              NaN              NaN   
2       -1.000000          -0.317073        -0.317073        -0.938950   
3             NaN                NaN              NaN              NaN   
4       -0.863874          -0.317073        -0.414634        -0.979069   

   BLAST_MEAN  CALCIUM_MEAN  CREATININ_MEAN  FFA_MEAN  GGT_MEAN  GLUCOSE_MEAN  \
0         NaN           NaN             NaN       NaN       NaN           NaN   
1         NaN           NaN             NaN       NaN       NaN           NaN   
2        -1.0      0.183673       -0.868365 -0.742004 -0.945093     -0.891993   
3         NaN           NaN             NaN       NaN       NaN           NaN   
4        -1.0      0.326531       -0.926398 -0.859275 -0.669393     -0.891993   

   HEMATOCRITE_MEAN  HEMOGLOBIN_MEAN  INR_MEAN  LACTATE_MEAN  LEUKOCYTES_MEAN  \
0               NaN              NaN       NaN           NaN              NaN   
1               NaN              NaN       NaN           NaN              NaN   
2          0.090147         0.109756 -0.932246      1.000000        -0.835844   
3               NaN              NaN       NaN           NaN              NaN   
4         -0.320755        -0.353659 -0.979925     -0.963023        -0.762843   

   LINFOCITOS_MEAN  NEUTROPHILES_MEAN  P02_ARTERIAL_MEAN  P02_VENOUS_MEAN  \
0              NaN                NaN                NaN              NaN   
1              NaN                NaN                NaN              NaN   
2        -0.914938          -0.868747          -0.170732        -0.704142   
3              NaN                NaN                NaN              NaN   
4        -0.643154          -0.868747          -0.365854        -0.230769   

   PC02_ARTERIAL_MEAN  PC02_VENOUS_MEAN  PCR_MEAN  PH_ARTERIAL_MEAN  \
0                 NaN               NaN       NaN               NaN   
1                 NaN               NaN       NaN               NaN   
2           -0.779310         -0.754601 -0.875236          0.234043   
3                 NaN               NaN       NaN               NaN   
4           -0.875862         -0.815951 -1.000000          0.574468   

   PH_VENOUS_MEA

É possivel perceber que existem muitas lugares em que o dado não está preenchido, e apresenta o "NaN".

* usar getdummies em age_percentil e 
* substituir os dados window por categoricos